Add color argument to button mixins

jeremyvii opened the pull request:
Fix #28899 by adding optional color arguments, for each button state, to `button-variant()` and `button-outline-variant()`. The color contrast function was moved to the default values of these mixins to reduce the need for conditions inside the mixin body.

To do:
- [x] Add documentation

jeremyvii committed:
Add color argument to button mixins

Add optional `$color` argument to `button-variant()` and 
`button-outline-variant()` for additional flexibility.

jeremyvii committed:
Merge branch 'master' into manual-btn-variant-color

MartijnCuppens commented:
This breaking change should be documented in the migration guide (probably under the Sass section).

jeremyvii committed:
Document changes to `button-variant()`

jeremyvii committed:
Merge branch 'manual-btn-variant-color' of github.com:jeremyvii/bootstrap into manual-btn-variant-color

mdo commented on scss/mixins/_buttons.scss:
```diff
@@ -3,35 +3,35 @@
 
-@mixin button-variant($background, $border, $hover-background: darken($background, 7.5%), $hover-border: darken($border, 10%), $active-background: darken($background, 10%), $active-border: darken($border, 12.5%)) {
-  color: color-yiq($background);
+@mixin button-variant($background, $border, $color: color-yiq($background), $hover-background: darken($background, 7.5%), $hover-border: darken($border, 10%),  $hover-color: color-yiq($hover-background), $active-background: darken($background, 10%), $active-border: darken($border, 12.5%), $active-color: color-yiq($active-background)) {
```
Could we start to reformat our mixins with all these parameters? Starting to feel difficult to read :).

```suggestion
@mixin button-variant(
  $background,
  $border, $color: color-yiq($background),
  $hover-background: darken($background, 7.5%),
  $hover-border: darken($border, 10%), 
  $hover-color: color-yiq($hover-background),
  $active-background: darken($background, 10%),
  $active-border: darken($border, 12.5%),
  $active-color: color-yiq($active-background)
) {
```

jeremyvii replied:
I agree, that would be much easier to read. Was the `$color` argument, on line 8, intentional on the same line as `$border`? Or should it be moved to it's own newline?

MartijnCuppens replied:
> Or should it be moved to it's own newline?

Yeah, it should (I've updated @mdo 's comment). We should also do this with the outlined buttons.

jeremyvii committed:
Move mixin arguments to their own line

Co-Authored-By: Mark Otto <otto@github.com>

jeremyvii committed:
Merge branch 'master' into manual-btn-variant-color

jeremyvii committed:
Move mixin arguments to their own line

MartijnCuppens commented:
The parameters in the button mixin have changed, this should be documented in migration guide.

XhmikosR reviewed (dismissed):
Please address MartijnCuppens's comment

XhmikosR committed:
Merge branch 'master' into manual-btn-variant-color

jeremyvii committed:
Document changes to `button-outline-variant()`

jeremyvii commented:
I updated the migration guide to include the changes to `button-outline-variant()` along with what I had added for `button-variant()`. Was this all I needed to document, or was there somewhere else I needed to address?

XhmikosR committed:
Merge branch 'master' into manual-btn-variant-color

jeremyvii committed:
Merge branch 'master' into manual-btn-variant-color

XhmikosR commented:
@MartijnCuppens LGTY now?

MartijnCuppens reviewed (approved):
Looks fine, thanks!
